Documentation/VSCodeConfiguration.md

```diff
@@ -4,20 +4,57 @@ Visual Studio Code does not work optimally for Serenity development, and there's
 
 The WSL Remote extension allows you to use VSCode in Windows while using the normal WSL workflow. This works surprisingly well, but for code comprehension speed you should put the Serenity directory on your WSL root partition.
 
-## Code comprehension
+- [Visual Studio Code Project Configuration](#visual-studio-code-project-configuration)
+  - [Code comprehension](#code-comprehension)
+    - [clangd](#clangd)
+      - [GCC](#gcc)
+      - [Clang](#clang)
+    - [Microsoft C/C++ tools](#microsoft-cc-tools)
+  - [DSL syntax highlighting](#dsl-syntax-highlighting)
+  - [Formatting](#formatting)
+  - [Settings](#settings)
+  - [Customization](#customization)
+    - [Custom Tasks](#custom-tasks)
+    - [License snippet](#license-snippet)
 
-Both C++ comprehension tools listed below report fake errors.
+## Code comprehension
 
 ### clangd
 
-The official clangd extension can be used for C++ comprehension. You'll have to use the following .clangd:
+The clangd extension (`llvm-vs-code-extensions.vscode-clangd`) can be used for C++ comprehension.
+
+You'll have to use the following `.clangd` file for the toolchain you're using:
+
+#### GCC
 
 ```yaml
 CompileFlags:
-  CompilationDatabase: Build/i686
+   Add:
+      [
+         -D__serenity__,
+         -isystem../../Toolchain/Local/i686/i686-pc-serenity/include/c++/11.2.0,
+         -isystem../../Toolchain/Local/i686/i686-pc-serenity/include/c++/11.2.0/bits,
+         -isystem../../Toolchain/Local/i686/i686-pc-serenity/include/c++/11.2.0/i686-pc-serenity,
+      ]
+   CompilationDatabase: Build/i686
+
+```
+
+#### Clang
+
+```yaml
+CompileFlags:
+   Add:
+      [
+         -D__serenity__,
+         -isystem../../Toolchain/Local/clang/include/c++/v1,
+         -isystem../../Toolchain/Local/clang/include/x86_64-pc-serenity/c++/v1,
```

Comment:
I believe these aren't required

Reply:
BertalanD provided me that snippet in the Discord conversation
Unless proven wrong that's there I suppose

Reply:
These *are* needed, unless you want to use the Serenity-aware `clangd` that's built as part of the toolchain. Neither the author of this PR, nor I have experience with setting that up though, so @Hendiadyoin1, you'd have to provide the relevant information for that. It's such a niche use case though, so it shouldn't hold up this PR IMO.

Reply:
I have also not used the serenity-built clangd so both options should be available